berkdb/mp/mp_fopen.c

```diff
@@ -799,8 +802,10 @@ __memp_fopen(dbmfp, mfp, path, flags, mode, pagesize)
 	 * create a new entry for the current request.
 	 */
 	R_LOCK(dbenv, dbmp->reginfo);
-	for (mfp = SH_TAILQ_FIRST(&mp->mpfq, __mpoolfile);
-	    mfp != NULL; mfp = SH_TAILQ_NEXT(mfp, q, __mpoolfile)) {
+	fileid_mpf = hash_find(mp->mpfhash, dbmfp->fileid);
+	if (!fileid_mpf)
```

Comment:
in spirit of making code slightly more readable, how about 
if(field_mp)
  for_each....
  
instead of goto?